feat(curriculum): add binary search tree (#52371)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-tree-traversal-by-building-a-binary-search-tree/655a227e57aabb25d1f9c987.md

---
id: 655a227e57aabb25d1f9c987
title: Step 19
challengeType: 20
dashedName: step-19
---

# --description--

Now, it's time to work on the search functionality.

> Create a method named `search` inside the `BinarySearchTree` class. Give the `search` method two parameters: `self` and `key`.

Reviewer: ```suggestion
Define a method named `search` inside the `BinarySearchTree` class. Give the `search` method two parameters: `self` and `key`.
```
I would suggest replacing "create" with "define" to be consistent with the hint.

---
id: 655a227e57aabb25d1f9c987
title: Step 20
challengeType: 20
dashedName: step-20
---

# --description--

Now, it's time to work on the search functionality.

> Create a function named `search` inside the `BinarySearchTree` class. The search method would take two parameters: `self` and `key`.

Reviewer 2: ```suggestion
Create a method named `search` inside the `BinarySearchTree` class. Give the `search` method two parameters: `self` and `key`.
```

> `self` refers to the instance of the `BinarySearchTree` class on which the method is called.
> `key` is the value that the user wants to search for in the BST.

Reviewer 2: ```suggestion
```
